npm/postinstall.ts

```diff
@@ -0,0 +1,211 @@
+import path from "path";
+import request from "request";
+import tar from "tar";
+import unzipper from "unzipper";
+import { URL } from "url";
+
+/**
+ * A map of Node's `process.arch` to Golang's `$GOARCH` equivalent.
+ */
+const archMap: Partial<Record<typeof process.arch, string>> = {
+  arm64: "arm64",
+  x64: "x86_64",
+};
+
+/**
+ * Known extensions that we package our binaries with.
+ *
+ * In an enum so that we can be type-safe with Record<> and ensure we're always
+ * handling any extensions we add here.
+ */
+enum KnownExtension {
+  Tar = ".tar.gz",
+  Zip = ".zip",
+}
+
+/**
+ * A map of Node's `process.platform` to Golang's `$GOOS` equivalent and the
+ * extension of the packaged binary we provide.
+ */
+const platformMap: Partial<
+  Record<
+    typeof process.platform,
+    { platform: string; extension: KnownExtension }
+  >
+> = {
+  linux: { platform: "linux", extension: KnownExtension.Tar },
+  darwin: { platform: "darwin", extension: KnownExtension.Tar },
+  win32: { platform: "windows", extension: KnownExtension.Zip },
+};
+
+/**
+ * Fetches the supposed URL of the binary on GitHub for this version.
+ */
+async function getBinaryUrl(): Promise<URL> {
+  const version = process.env.npm_package_version?.trim();
+
+  if (!version) {
+    throw new Error("Could not find package version to install binary");
+  }
+
+  const { arch, platform } = getArchPlatform();
+
+  const targetUrl = new URL(
+    `https://github.com/inngest/inngest/releases/download/v${version}/inngest_${version}_${platform.platform}_${arch}${platform.extension}`
+  );
+
+  return targetUrl;
+}
+
+/**
+ * Return the platform and architecture included in the supposed binary name
+ * from `$GOOS` and `$GOARCH`.
+ *
+ * @throws when the arch or platform are not supported by this script.
+ */
+function getArchPlatform() {
```

Comment:
It'd be cool in the future if this used our `.goreleaser.yml` file to figure out what is supported.